### PR TITLE
Update for bevy ecs v2

### PR DESCRIPTION
**Pull request by rparrett**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -22,8 +22,8 @@ impl Plugin for TiledMapPlugin {
             .add_event::<MapReadyEvent>()
             .add_system(process_loaded_tile_maps.system());
 
-        let resources = app.resources();
-        let mut render_graph = resources.get_mut::<RenderGraph>().unwrap();
-        render_graph.add_tile_map_graph(resources);
+        let world = app.world_mut();
+
```
Comment (dmtaub) [inline src/lib.rs:26]: well! that got easier.. :)

File: src/map.rs [hunk 2]
```diff
@@ -567,15 +581,17 @@ impl Object {
         }
         .with(map_handle)
         .with(self.clone())
+        .current_entity()
+        .unwrap()
```
Comment (dmtaub) [inline src/map.rs:585]: Just a couple comments here:
- I've been trying to get away from naked unwraps, personally :) 
- I liked the API where we could chain, just like a normal spawn...
  - can you do this in your consuming application?
  - did you have trouble with the lifetime arguments?

**Review by dmtaub — COMMENTED** [review]
Thanks so much for figuring out compatibility with ECS v2 !
I like the changes your linter made about 90% of the time, too..

One thing that changed in this PR is the API for spawn(), which I'd like to try to put back so it can be chainable, since I know we eventually want an option for calling spawn manually (the automatic spawning of objects makes it hard to deal with timing). Also, removal of map means that it will panic if that operation fails for some reason.

I'll see if I can do that, and if so merge this plus that extra commit -- thanks again for figuring this out!

File: src/map.rs (hunk 2 above)
Comment (rparrett) [reply src/map.rs:585]: Yep, stumped by lifetime mismatch errors.

I could take another stab at it.

File: src/lib.rs (hunk 1 above)
Comment (rparrett) [reply src/lib.rs:26]: Apologies for the cargo fmt.

Happy to redo this with fmt-on-save disabled if you prefer.

**Pull request merged**